sqlmesh/core/renderer.py

```diff
@@ -434,7 +442,9 @@ def _optimize_query(self, query: exp.Subqueryable) -> exp.Subqueryable:
         original = query
         failure = False
         missing_deps = set()
-        all_deps = d.find_tables(query, dialect=self._dialect) - {self._model_name}
+        all_deps = d.find_tables(
+            query, default_catalog=self._default_catalog, dialect=self._dialect
+        ) - {self._model_name, self._model_fqn}
```

Comment:
this should only remove model_fqn, model_name should not be present

Reply:
again old code thanks